### PR TITLE
Adding a check to see if a ticket number is for Zendesk instead of internal

### PR DESCRIPTION
**Pull request by victoriagrey**

### CODE REVIEW

File: scripts/sg-ticket-link.coffee [hunk 1]
```diff
@@ -28,5 +28,9 @@ module.exports = (robot) ->
     if isNaN(ticket_number)
       return
     
-    shotgun_url = process.env.HUBOT_SHOTGUN_URL
-    msg.send "#{shotgun_url}/detail/Ticket/#{ticket_number}"
+    if msg.message.match(/ZD#(\d+)/)
+      zendesk_url = process.env.HUBOT_ZENDESK_TICKET_URL
+      msg.send "#{zendesk_url}#{ticket_number}"
```
Comment (nemoDreamer) [inline scripts/sg-ticket-link.coffee]: Can't you just make sg-ticket-link shut up when the match is `ZD#`, and have the Zendesk script handle it? That way, you don't muddy up concerns.

Or better: change the regexp for this to force leading white-space: `/\s+#(\d+)/`

Comment (victoriagrey) [inline scripts/sg-ticket-link.coffee]: I don't think we have the Zendesk script in this repo right now... I'm assuming there's an upstream somewhere I'm not looking at :p. As for the HipChat supported integration-- it doesn't currently support the Mac Desktop app.

As for whitespace-- what if we want to add other types in the future? shouldn't we be as specific as possible and not make assumptions?

**Comment by nemoDreamer** [comment]
I suggest:
- change this script's RegExp to expect a leading whitespace char.
- create `sg-zendesk-ticket`, w/ the `/ZD#(\d+)/` RegExp, as a copy of this one.

**Comment by victoriagrey** [comment]
Alrighty-- made the zendesk changes their own script, and added a whitespace qualifier to the sg-ticket-link regex

File: scripts/sg-zendesk-ticket.coffee [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# Author:
+#   neilvictorgrey
+
+module.exports = (robot) ->
+  githubIgnoreUsers = process.env.HUBOT_SG_ISSUE_LINK_IGNORE_USERS
+  if githubIgnoreUsers == undefined
+    githubIgnoreUsers = "hubot"
+
+  robot.hear /\s+ZD#(\d+)/i, (msg) ->
+    return if msg.message.user.name.match(new RegExp(githubIgnoreUsers, "gi"))
```
Comment (nemoDreamer) [inline scripts/sg-zendesk-ticket.coffee:27]: Love it!

**Pull request merged**